crates/yulgen/src/runtime/functions/abi.rs

```diff
@@ -133,47 +146,55 @@ pub fn decode_component<T: AbiEncoding>(typ: &T, location: AbiDecodeLocation) ->
 pub fn decode_component_uint(size: usize, location: AbiDecodeLocation) -> yul::Statement {
     let func_name = abi_names::decode_component_uint(size, location);
     let decode_expr = load_word(expression! { ptr }, location);
+    let check_padding = abi_operations::check_left_padding((32 - size) * 8, expression! { return_val });
 
     function_definition! {
          function [func_name](head_start, offset) -> return_val {
             (let ptr := add(head_start, offset))
             (return_val := [decode_expr])
+            [check_padding]
          }
     }
 }
 
 pub fn decode_component_int(size: usize, location: AbiDecodeLocation) -> yul::Statement {
     let func_name = abi_names::decode_component_int(size, location);
     let decode_expr = load_word(expression! { ptr }, location);
+    // let check_padding = abi_operations::check_left_padding((32 - size) * 8, expression! { return_val });
```

Comment:
@cburgdorf I think our signed integers might have some problems..

This check failed, since all signed integers are encoded as if they are an `i256`, which means, for example, that an `i8` will use the leftmost bit in the word for signedness, as opposed to the *8th* bit from the right. Our decoding functions should account for this by moving the signdness bit to the appropriate spot and validating that all of the other bits are zero. They aren't, though, and the existing tests seem to work fine, which leads me to believe that all signed integers are effectively being treated like `i256`s. If this is the case, certain data operations will erase signs. For example, storing a negative `i8` value in an array and then reading it will yield a different (positive) number, since the sign was erased.

Reply:
> Our decoding functions should account for this by moving the signdness bit to the appropriate spot

Well, maybe not actually. It might be better to treat all signed integers like `i256` values when they're on the stack, and only move the sign when reading and writing to memory/storage. This way we can still make use of the existing signed math functions.

Reply:
>since all signed integers are encoded as if they are an `i256`

Ah, that's right.

>Well, maybe not actually. It might be better to treat all signed integers like i256 values when they're on the stack, and only move the sign when reading and writing to memory/storage

Yep, that makes sense to me :+1: If you aren't going to address this in this PR we should create an issue and flag it with `mvp-required`

Reply:
I'll write an issue